prymer/primer3/primer3_parameters.py

```diff
@@ -62,19 +62,61 @@ class stores user input for internal probe design and maps it to the correct Pri
 
 ```
 """
-
-import warnings
+import typing
+from abc import ABC, abstractmethod
 from dataclasses import dataclass
 from dataclasses import fields
 from typing import Any
 from typing import Optional
 
-from prymer.model import MinOptMax
+from prymer.model import MinOptMax, Span
 from prymer.primer3.primer3_input_tag import Primer3InputTag
+from prymer.primer3.primer3_task import Primer3TaskType
+
+
+class Primer3Parameters(ABC):
+    target: Span
+    task: Primer3TaskType
```

Comment:
I think I would prefer to keep these separate and just replace `Primer3Input` with providing 3 args to `primer3.design()` so that the same parameters can be re-used across many calls for many targets without having to `dataclasses.replace(param, target=next_target)` each time.